### ISSUE TITLE
[Bug]: JEI breaks Fluid handling recipes

### ISSUE THREAD

**Issue by TheonlyTazz** [mezz/JustEnoughItems#2936]
### Steps to Reproduce the Bug

in the latest v10.x.x.xxx versions fluid recipes are broken and it doesnt show thermal expansions bottler recipes in JEI anymore

in the latest v9.7.1.232 the recipes are visible but arent handled correctly in Refined Storage and Applied Energistics:

1. Create Refined Storage/Applied Energistics setup
2. Open Pattern Grid/Terminal
3. Try to add a recipe from the Thermal Bottler (Example Milk Bucket)
4. Fluid is actually in the output slot of the Pattern

![image](https://user-images.githubusercontent.com/3908847/182824843-3cba13ab-1c5f-4c79-906f-a9984b099593.png)

### Expected Behavior

have the fluid be in the input side

### Actual Behavior

Fluid is in output slot

### Mod Pack URL (Optional)

https://feed-the-beast.com/modpack/95_ftb_presents_direwolf20_1_18

### Mod Pack Version (Optional)

v.1.8

### Extra Notes (Optional)

This issue is tracked on the FTB Modpack Issue Tracker
https://github.com/FTBTeam/FTB-Modpack-Issues/issues/1157

**Comment by mezz** [comment]
Thanks for the report!
This is a known incompatibility with the JEI 10 alpha, please use JEI 9 until it is resolved.

**Comment by mezz** [comment]
For the second issue, is this happening in the same way for both mods when using JEI 9.7?

**Comment by TheonlyTazz** [comment]
yes, both pattern terminals place the actual liquid in the output slots instead of placing them in the input slots

**Comment by mezz** [comment]
This was caused by Thermal Expansion’s JEI integration and has been fixed upstream:
* Fluid input/output role issue: https://github.com/CoFH/ThermalExpansion/commit/ba4c7493f859c80cf4375eb3053909b41d00945e
* JEI 10 fluid API compatibility issue: https://github.com/CoFH/ThermalExpansion/commit/3983f907e96a2a7f74172e771b8e354a53c5e2f6
The linked FTB report was also confirmed fixed, so I’m closing this issue. Thanks!
